BizHawk.Emulation.Common/Sound/Utilities/BlipBuffer.cs

```diff
@@ -147,7 +147,7 @@ public void EndFrame(uint clockDuration)
 
 		public int SamplesAvailable()
 		{
-			if (PlatformLinkedLibSingleton.RunningOnUnix) return 0; // temporary workaround
+			if (PlatformLinkedLibSingleton.RunningOnUnix) return 882; // temporary workaround

```

Comment:
So this is *i think* consistently true for ZXHawk. Probably not for other cores that utilise blipbuffer

Reply:
I was hoping Sappharad or someone else would point me to either a source for blip_buf.so or build instructions because I have no idea how to make such a file. If that can't be done and this value does change with the core, maybe I could add that as a parameter, as `SamplesAvailable` here is actually wrapping the library.

Reply:
I’ll look tonight.

Rough steps:
Find C++ IDE for Linux
Create library project
Point project to source files
Compile

Normal people would probably use command line tools to do it instead of an IDE. But many Linux IDEs probably generate command line makefiles anyway.

Reply:
The blip_buf code is actually in the root of the repo. Modifying the make to output an so instead of a dll and running the makefile built fine. Completely untested as I only have the Linux Subsystem for Windows set up at the moment though.

Reply:
> Modifying the makefile to output an so instead of a dll and running make built fine.

I really don't think that's how it works. One has to use mingw to build a .dll out of code that builds .so if you just compile on linux.

Reply:
Windows subsystem for Linux uses native Ubuntu binaries. The .so built by it should be fine for Linux. 

As long as it’s called libblip_buf.so and located in the same place as the windows DLL mono will automatically use it. No Cygwin, that would make a windows DLL and he won’t want that.

Reply:
I mean it's not achieved by hacking the makefile, is it?

Reply:
When I said modified the makefile I should have been more clear. I literally just modified it to name the output .so rather than .dll. The makefile within the directory literally just calls gcc with a few specific options

Reply:
I went and set up a VM to test it. Got it all built based on the branch but it's yelling at me about not being able to find DX9, VC++ 2010, 2012, and 2015. I'm clearly missing something. @YoshiRulz

Reply:
It's looking for prereqs. Are these even available on linux?

Reply:
I assumed the Mono branch would deal with that but it appears it doesn't. There may be some dependency besides just mono required? @YoshiRulz Could probably elaborate

Reply:
Yeah, I’m curious how that’s supposed to work too. I ran into that a few weeks back when I pulled the latest code into my fork and had to replace a bunch of if(runningOnUnix) checks with #if WINDOWS again. Someone mentioned in IRC that it was for WINE so I assumed that was valid.

Reply:
Wine doesn't run 2.x bizhawk at all. There are some insane problems with dependencies. It runs 1.x though, and I dunno if that was intended wither when those edits were made. 1.x running in wine was found out like this year, but never seriously verified or examined.

Reply:
@NarryG @Sappharad It doesn't need WINE, just mono! You'll need WINE *installed* to get those missing libraries, but it's simply `LD_LIBRARY_PATH="/usr/lib/wine" mono EmuHawk.exe` to run it. Also I've forked `master` not `mono-portable` in case that was unclear.

Reply:
I took a look at it, and the changes @NarryG made seem to be all that is needed. Just update makefile to change the output name from blip_buf.dll to libblip_buf.so and run make at the terminal. It looks like the -shared flag is the key to making an .so and it's already set.

For some reason, GitHub isn't letting me attach it directly to this comment, so here's a compiled build that I made with Ubuntu:
https://www.sappharad.com/junk/libblip_buf.so.zip

Regarding WINE, I think the problem there is that .NET stuff doesn't really work correctly at all in 64-bit WINE. I just tried it myself and the dependencies didn't give me trouble, I created a 64-bit bottle and those installed fine. But the BizHawk blows up trying to use classes that are supposed to be part of the .NET Framework. I guess it'll probably be another year or so before .NET is working properly.

Reply:
I attempted running it that way but it didn't work. Might have been a mistake on my side.

Reply:
Worked for me, though the copy I built was different from @Sappharad's.